pkg/certifier/clearlydefined/clearlydefined.go

```diff
@@ -48,38 +48,43 @@ const (
 var ErrOSVComponentTypeMismatch error = errors.New("rootComponent type is not []*root_package.PackageNode")
 
 type cdCertifier struct {
+	cdHTTPClient *http.Client
 }
 
-// NewClearlyDefinedCertifier initializes the the cdCertifier
+// NewClearlyDefinedCertifier initializes the cdCertifier
 func NewClearlyDefinedCertifier() certifier.Certifier {
-	return &cdCertifier{}
+	return &cdCertifier{
+		cdHTTPClient: clients.NewClearlyDefinedClient(),
+	}
+}
+
+func NewClearlyDefinedHTTPClient() *http.Client {
+	return clients.NewClearlyDefinedClient()
 }
 
 // GetPkgDefinition uses the coordinates to query clearly defined for license definition
-func GetPkgDefinition(ctx context.Context, coordinate *coordinates.Coordinate) (*attestation.Definition, error) {
+func GetPkgDefinition(ctx context.Context, client *http.Client, coordinate *coordinates.Coordinate) (*attestation.Definition, error) {
 	logger := logging.FromContext(ctx)
 
 	url := fmt.Sprintf("https://api.clearlydefined.io/definitions/%s/%s/%s/%s/%s", coordinate.CoordinateType, coordinate.Provider,
 		coordinate.Namespace, coordinate.Name, coordinate.Revision)
 
-	req, err := http.NewRequest("GET", url, nil)
+	req, err := http.NewRequestWithContext(ctx, "GET", url, nil)
 	if err != nil {
 		return nil, fmt.Errorf("failed to generate new request with error: %w", err)
 	}
-	resp, err := version.UATransport.RoundTrip(req)
+	resp, err := client.Do(req)
 	if err != nil {
 		return nil, fmt.Errorf("failed to get response from clearly defined API with error: %w", err)
 	}
 	defer resp.Body.Close()
 
 	if resp.StatusCode != http.StatusOK {
 		if resp.StatusCode == http.StatusNotFound {
-			// log the error when not found but don't return the error to continue the loop
```

Comment:
Why remove these comments?

Reply:
I removed the comments while tweaking the code, I forgot to add them back, thanks!

Reply:
go ahead and add them back

Reply:
Yeah... now I added them back... I thought I added them back when I replied to you